@@ -, +, @@ 
---
 shell/ev-application.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)
--- a/shell/ev-application.c	
+++ a/shell/ev-application.c	
@@ -1044,8 +1044,8 @@ ev_application_startup (GApplication *gapplication)
           "win.select-page",            "<Ctrl>L", NULL,
           "win.go-backwards",           "<Shift>Page_Up", NULL,
           "win.go-forward",             "<Shift>Page_Down", NULL,
-          "win.go-next-page",           "n", NULL,
-          "win.go-previous-page",       "p", NULL,
+          "win.go-next-page",           "n", "<Ctrl>Page_Down", NULL,
+          "win.go-previous-page",       "p", "<Ctrl>Page_Up", NULL,
           "win.go-back-history",        "<alt>P", "Back", NULL,
           "win.go-forward-history",     "<alt>N", "Forward", NULL,
           "win.sizing-mode::fit-page",  "f", NULL,
-- 
