--- evince-3.28.2/backend/comics/ev-archive.c
+++ evince-3.28.2/backend/comics/ev-archive.c
@@ -55,7 +55,7 @@ ev_archive_finalize (GObject *object)
 	case EV_ARCHIVE_TYPE_ZIP:
 	case EV_ARCHIVE_TYPE_7Z:
 	case EV_ARCHIVE_TYPE_TAR:
-		g_clear_pointer (&archive->libar, archive_free);
+		g_clear_pointer (&archive->libar, archive_read_free);
 		break;
 	default:
 		break;
@@ -278,7 +278,7 @@ ev_archive_get_entry_is_encrypted (EvArc
 	case EV_ARCHIVE_TYPE_7Z:
 	case EV_ARCHIVE_TYPE_TAR:
 		g_return_val_if_fail (archive->libar_entry != NULL, -1);
-		return archive_entry_is_encrypted (archive->libar_entry);
+		return FALSE;
 	}
 
 	return FALSE;
@@ -336,7 +343,7 @@ ev_archive_reset (EvArchive *archive)
 	case EV_ARCHIVE_TYPE_ZIP:
 	case EV_ARCHIVE_TYPE_7Z:
 	case EV_ARCHIVE_TYPE_TAR:
-		g_clear_pointer (&archive->libar, archive_free);
+		g_clear_pointer (&archive->libar, archive_read_free);
 		libarchive_set_archive_type (archive, archive->type);
 		break;
 	default:
--- evince-3.28.2/configure.ac
+++ evince-3.28.2/configure.ac
@@ -698,7 +698,7 @@ AC_ARG_ENABLE(comics,
 	[enable_comics=yes])
 
 if test "x$enable_comics" = "xyes"; then
-   LIBARCHIVE_REQUIRED=3.2.0
+   LIBARCHIVE_REQUIRED=3.1.2
    PKG_CHECK_MODULES(LIBARCHIVE, libarchive >= $LIBARCHIVE_REQUIRED,enable_comics=yes,enable_comics=no)
 
    if test "x$enable_comics" = "xyes"; then
